htdocs/product/card.php

```diff
@@ -813,6 +816,9 @@
 					$object->fk_default_bom = 0;
 				}
 
+				// managed_in_stock
+				$object->stockable_product   = GETPOSTISSET('stockable_product');
```

Comment:
This line looks strange.
if stockable_product = 0,
you wil get
$object->stockable_product = true.

i think what you want to do is 
$object->stockable_product = GETPOSTINT('stockable_product');

Reply:
No, since stockable_product is a checkbox. 
- When not checked, $_POST['stockable_product'] is not set
- When checked, $_POST['stockable_product'] is equal to string 'on'. As a consequence, `GETPOSTINT('stockable_product') == 0`.

The exact way to deal with it is `$object->stockable_product = (int) GETPOST('stockable_product') == 'on'`. But it makes useless operations and it does not harm security to `$object->stockable_product   = GETPOSTISSET('stockable_product');` .

I will cast it to (int).